source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,462 @@
+switch.
+
+## Prerequisites
+
+LLDP has to be enabled on the switch connected to the relevant oVirt hosts.
+
+## Limitations
+
+* There is a delay from receiving the LLDP data on the oVirt host, propagating
+  the data to the oVirt Engine and presenting the data to the user.
+  Whatever value we present over REST may be out of date.
+
+## Benefit to oVirt
+
+Information from LLDP provided by oVirt engine's REST-API will support the
+integration of oVirt in other software.
+
+Information from LLDP provided by the UI might help the user to avoid and
+locate manual host network configuration errors caused by a mismatch of switch
+port configuration.
+
+## Requirements
+
+* The following TLVs have to be available in oVirt Engine's REST API:
+  * Chassis ID TLV (Type = 1)
+  * Port ID TLV (Type = 2)
+  * Port Description TLV (Type = 4), with a value of up to 2000 chars
+  * System Name TLV (Type = 5)
+  * System Description TLV (Type = 6)
+  * Management Address TLV (Type = 8)
+  * Port VLAN Id ( Type = 127, Oui = 0x0080c2 (802.1), Subtype = 1)
+  * VLAN name (Type = 127, Oui = 0x0080c2 (802.1), Subtype = 3)
+
+## Open Questions
+
+* How is a Port Description with a length than more than 512 chars represented
+  in LLDP?
+* Should oVirt Engine provide the age of the data?
+* Should oVirt Engine hide expired data?
+* Is there an upper limit of the acceptable age of the data?
+
+## Approach
+
+Steps to present the LLDP information:
+
+1. [Receive LLDP information on the Ethernet cable on the host](#vdsm-implementation-options)
+2. [Cache the LLDP information on the host](#vdsm-implementation-options)
+3. [Transport the LLDP information to oVirt Engine](#communication-of-vdsm-and-ovirt-engine)
+4. Present LLDP information:
+  * [On REST-API](#rest-api)
+  * [On Web-UI](#follow-up-features)
+  * [For oVirt Engine internal use](#follow-up-features)
+
+If the LLDP information is used internally in oVirt Engine or in Web-UI, it
+would be useful to cache the information in the data base.
+
+To meet the requirements of this feature, there is no need to store the
+information in the data base, because the information could be queried when a
+REST request is processed processed. This way the age of the information is
+kept as minimal as possible.
+
+Thus we opt to avoid the DB completely. When Engine is asked about LLDP data of
+a host, it would contact the specific host synchronously in order to provide an
+answer.
+
+## Gathering the LLDP data from the hosts
+
+There are multiple alternatives for oVirt Engine to gather the LLDP data from
+the hosts:
+
+* Extend VDSM to provide the LLDP data
+
+* Use [Ansible's lldp module][3]
+
+Because the Ansible module is flagged as preview and requires lldpd, which is
+in only in EPEL repository for RHEL, VDSM has to be extended.
+
+## VDSM implementation options
+
+to be imported from [EdDev/ovirt-devwiki](https://github.com/EdDev/ovirt-devwiki/blob/master/design/network/lldp/lldp.md)
```

Comment:
There's hardly anything there. Specifically, I'm wondering if we'll support CDP.